packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { Column, RowData } from 'react-data-grid';
+
+enum RuleType {
+  Number = 1,
+  Range = 2,
+  GreaterThen = 3,
+  LessThen = 4
+}
+
+type Rule =
+  | { type: RuleType.Range; begin: number; end: number }
+  | { type: RuleType.GreaterThen | RuleType.LessThen | RuleType.Number; value: number };
+
+interface ChangeEvent {
+  filterTerm: Rule[] | null;
+  column: Column;
+  rawValue: string;
+  filterValues: typeof filterValues;
+}
+
+interface Props {
+  column: Column;
+  onChange(event: ChangeEvent): void;
+}
+
+export default function NumericFilter({ column, onChange }: Props) {
+  /** Validates the input */
+  function handleKeyPress(event: React.KeyboardEvent<HTMLInputElement>) {
+    const result = /[><,0-9-]/.test(event.key);
+    if (result === false) {
+      event.preventDefault();
+    }
+  }
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const value = event.target.value;
+    const filters = getRules(value);
+    onChange({
+      filterTerm: filters.length > 0 ? filters : null,
+      column,
+      rawValue: value,
+      filterValues
+    });
+  }
+
+  const inputKey = `header-filter-${column.key}`;
+  const columnStyle: React.CSSProperties = {
+    float: 'left',
+    marginRight: 5,
+    maxWidth: '80%'
+  };
+  const badgeStyle: React.CSSProperties = {
+    cursor: 'help'
+  };
+
+  const tooltipText = 'Input Methods: Range (x-y), Greater Then (>x), Less Then (<y)';
+
+  return (
+    <div>
+      <div style={columnStyle}>
+        <input
+          key={inputKey}
+          placeholder="e.g. 3,10-15,>20"
+          className="form-control input-sm"
+          onChange={handleChange}
+          onKeyPress={handleKeyPress}
+        />
+      </div>
+      <div className="input-sm">
+        <span className="badge" style={badgeStyle} title={tooltipText}>?</span>
+      </div>
+    </div>
+  );
+}
+
+
+function filterValues(row: RowData, columnFilter: { filterTerm: { [key in string]: Rule } }, columnKey: string) {
+  if (columnFilter.filterTerm == null) {
+    return true;
+  }
+
+  // implement default filter logic
+  const value = parseInt(row[columnKey] as string, 10);
+  for (const ruleKey in columnFilter.filterTerm) {
```

Comment:
`for (const rule of columnFilter.filterTerm)`

Reply:
It's not an iterable.

Reply:
Ah, I got confused with `filterTerm: Rule[] | null;`